ios/PresentationLayer/Recipes/Sources/Recipes/Books/BooksViewModel.swift

```diff
@@ -10,48 +10,75 @@ import SwiftUI
 import UIToolkit
 
 final class BooksViewModel: BaseViewModel, ViewModel, ObservableObject {
-    
+
     // MARK: Dependencies
     private weak var flowController: FlowController?
-    
+
     @Injected private(set) var getBooksUseCase: GetBooksUseCase
     @Injected private(set) var refreshBooksUseCase: RefreshBooksUseCase
 
     init(flowController: FlowController?) {
         self.flowController = flowController
         super.init()
+
+        // start book observing
+        onIntent(Intent.startBookObserving)
     }
-    
+
     // MARK: Lifecycle
-    
+
     override func onAppear() {
         super.onAppear()
-        executeTask(Task { await loadBooks() })
+        onIntent(Intent.refreshBooks)
     }
-    
+
     // MARK: State
-    
+
     @Published private(set) var state: State = State()
 
     struct State {
-        var isLoading: Bool = false
+        var isLoading: Bool = true
         var books: [Book] = []
     }
-    
+
     // MARK: Intent
+
     enum Intent {
+        case startBookObserving
```

Comment:
From the MVI perspective, this seems wrong to me. The intent should represent only the actions that a user can do in a view. I would prefer to call `executeTask(Task { await observeBooks() })` in the init instead of doing it via Intent.